python/sparkdl/image/imageIO.py

```diff
@@ -118,89 +82,62 @@ def imageStructToArray(imageRow):
     :param imageRow: Row, must use imageSchema.
     :return: ndarray, image data.
     """
-    imType = imageType(imageRow)
+    imType = imageTypeByOrdinal(imageRow.mode)
     shape = (imageRow.height, imageRow.width, imageRow.nChannels)
     return np.ndarray(shape, imType.dtype, imageRow.data)
 
+def imageStructToPIL(imageRow):
+    imgType = imageTypeByOrdinal(imageRow.mode)
+    if imgType.dtype != 'uint8':
+        raise ValueError("Can not convert image of type " + imgType.dtype + " to PIL, can only deal with 8U format")
+    ary = imageStructToArray(imageRow)
+    if imgType.nChannels != 1:
```

Comment:
let's be more explicit and fix `==3`. What do you think?

Reply:
it can also be 4. I think it is specific as it is - the only time you do not need to change the order of channels is when there is only one channel